### ISSUE TITLE
MariaDB version not displayed correctly

### ISSUE THREAD

**Issue by christine-kaminski** [piwik/piwik#11934]
We are using Piwik vs 3.0.2 on CentOS 7 with MariaDB 10.0.31. It works fine so far. Unfortunately the system information in the settings show MySQL 5.5.5 instead. Though Piwik and PHP version are displayed correctly. Any idea how to fix this?

**Comment by christine-kaminski** [comment]
![piwik](https://user-images.githubusercontent.com/18280676/28969688-e7473cd0-7925-11e7-98bc-f87afb015f6a.PNG)

**Comment by mneudert** [comment]
__tl;dr:__ The version you are seeing is not "wrong".

You could set a custom version string from MariaDB 10.2.6 onwards ([see here](https://jira.mariadb.org/browse/MDEV-10541)) but not earlier. And just fixing the version displayed might break other things yet to be known...

As long as MariaDB stays reasonably compatible with MySQL there should be no problems.

-----

In detail this is a result of how MariaDB reports it's version. Take for example the following code:

```php
$db = new PDO('mysql:host=127.0.0.1', 'root', 'password');
echo $db->getAttribute(PDO::ATTR_SERVER_VERSION);
```

If you execute this on your server in an interactive PHP shell (`php -a`) you should receive something like `5.5.5-10.0.31-MariaDB` (an Ubuntu 16.04 installation I just looked this up in returned `5.5.5-10.0.29-MariaDB-0ubuntu0.16.04.1`).

After receiving this value the underlying `Zend_Db_Adapter_Pdo_Abstract` extracts and returns ([see here](https://github.com/piwik/piwik/blob/7e36c6db68ece71d4a270e4ccc361854df494140/libs/Zend/Db/Adapter/Pdo/Abstract.php#L394)) just the first part of it. That is where your `5.5.5` is coming from.

Even further down the rabbit hole you see the MariaDB source code with an explanation ([see here](https://github.com/MariaDB/server/blob/f6633bf058802ad7da8196d01fd19d75c53f7274/include/mysql_com.h#L42)) about why they did that. Also mentioning they manually patched their clients to just strip this version from getting printed.

**Comment by sgiehl** [comment]
@mneudert do you think there is something we could improve in version detection for MariaDB? Or would that require changes to `Zend_Db_Adapter_Pdo_Abstract`?

**Comment by mneudert** [comment]
As modifying external libraries is always a bad idea the most promising change seems be extending `Piwik\Db\Adapter\Pdo\Mysql` with something like a `getRawServerVersion`.

Changing `getServerVersion` would mess with the version requirement checks as we would not compare a "MySQL compatibility promise" but a totally different versioning scheme.

Having a custom display version could provide something really specific like `MariaDB 10.0.2 (MySQL 5.5.5 compatible)`.

**Comment by christine-kaminski** [comment]
Thanks guys!
